### PR TITLE
Preprocessing initial code for making training data

### LINKED ISSUE #1: Collect GFSv16 T2m forecast results
This is the first step to do bias correction for the GFSv16 products. In this task:

1. a utility will be created to generate a time serious of the GFSV16 2m temperature, this will be used as input to the ML model
2. collect the ERA5 2m temperature during the same period, the data will be used as truth to tune the ML model.

### PR DESCRIPTION
**Pull request by sadeghitabas**
## Description
This PR provide a utility for preparing ML4BC model inputs from 0.25 degree resolution GFS and ERA5 data. This python utility have two modules one for gfs info and the second one for era5 info. At this stage we are going to train ML4BC model for 2m temprature forecasts with spatial and temporal resolution of 0.25 degree and hourly, respectively. In summary, this utility provides functionalities including:
- Downloading ECMWF's ERA5 and NOAA GFS hourly forecasts,
- Extract relevant variables from grib2 data
- Reshape, rename and transform extracted data to desired format
- Merge the data through time with desired number of forecast time steps and save it in netcdf format
- Produce ERA5 and GFS 2m temperature for model training

Also, a brief notebook provided to review, sanity check, and plot the 2m temperature datasets. 

### Associated GraphCast Issue to close
- Closes https://github.com/NOAA-EMC/ML4BC/issues/1

### Input data
- [ ] No changes are expected to input data.
- [x] Changes are expected to input data:
  - [ ] New input data.
  - [x] Updated input data.

### Needed libraries

  - `python3`
  - `xarray`
  - `boto3`
  - `netCDF4`
  - `pygrib`
  - `cdsapi`
  - `matplotlib`
  - `ipywidgets`

### CODE REVIEW

File: era5_job.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+#SBATCH -J gfs_process
+#SBATCH -o /contrib/Sadegh.Tabas/ML4BC/era5_output.out
```
Comment (junwang-noaa) [inline era5_job.sh]: Is this job card working on aws cloud? I am curious if we can remove the hardcode path with personal directory, e.g. #SBATCH - oera5_output.out ?

Comment (sadeghitabas) [reply era5_job.sh]: I removed the personal directories. Yes if you look at /contrib/Sadegh.Tabas/ML4BC/ I did run the code (still running) and preparing the training data. Thanks

File: era5_job.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+source ~/.bashrc
+
+# Activate the Conda environment
+conda activate ml4bc
+
+# Navigate to the working directory
+cd /lustre/ML4BC
+
+# Run the Python script
+python3 preprocessing.py 20210323 20220101 -p era5 -o /contrib/Sadegh.Tabas/ML4BC/
```
Comment (junwang-noaa) [inline era5_job.sh]: Could this be: /contrib/$USER/ML4BC ?

File: preprocessing.py [hunk 3]
```diff
@@ -0,0 +1,240 @@
+'''
+Description
```
Comment (junwang-noaa) [inline preprocessing.py]: Can you add a brief section on what the file is doing under Description?

File: preprocessing.py [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+@uthor: Sadegh Sadeghi Tabas (sadegh.tabas@noaa.gov)
```
Comment (junwang-noaa) [inline preprocessing.py]: Author?

**Review by junwang-noaa — APPROVED** [review]

**Pull request merged**